### PR TITLE
First PR

### PR DESCRIPTION
**Pull request by sarthak-gothalyan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Material Select
+
+This library provides material design select functionality implemented using extended typescript
+
+## Download & Install
+
+Install using npm
+
+```npm i material-select```
```
Comment (HMOS-Developer) [inline README.md]: update this to 
npm i @ohos/material-select

File: package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "buildTool": "hvigor",
+    "directoryLevel": "project"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {
+    "@ohos/hvigor": "1.0.6",
+    "@ohos/hvigor-ohos-plugin": "1.0.6",
+    "@ohos/material-select": "file:material-select",
```
Comment (HMOS-Developer) [inline package.json]: remove library dependency from here

File: entry/package.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  "version": "1.0.0",
+  "ohos": {
+    "org": "huawei",
+    "buildTool": "hvigor",
+    "directoryLevel": "module"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {}
```
Comment (HMOS-Developer) [inline entry/package.json]: add library dependency here - "@ohos/material-select": "file:material-select",

File: entry/src/main/ets/MainAbility/pages/index.ets [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+						.maxLines(2)
+						.textOverflow({ overflow: TextOverflow.Ellipsis })
+				}
+				.width('80%')
+				.padding(5)
+				.borderWidth(1)
+				.borderRadius(4)
+			}
+			.width('80%')
+			.margin({ left: 10, right: 10, top: 10, bottom: 10 })
```
Comment (archanaatr) [inline entry/src/main/ets/MainAbility/pages/index.ets]: Mark all dimentions as vp

File: material-select/src/dotMenu.ets [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+				}, item => item.id)
+			}
+		}
+		.width(this.menuModel.getWidth())
+		.padding({ top: 7, bottom: 7 })
+		.borderRadius(this.menuModel.getBorderRadius())
+	}
+
+	build() {
+		Image($r('app.media.dot_menu'))
```
Comment (archanaatr) [inline material-select/src/dotMenu.ets]: all the images / resources used in the library needs to be placed in the resource folder within te library, pls move them from entry to library

File: material-select/src/dotMenu.ets [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+			.objectFit(ImageFit.Contain)
+			.backgroundColor(this.model.getBackgroundColor())
+			.bindMenu(this.MenuBuilder())
+			.margin(this.model.getMargin())
+	}
+}
+
+namespace DotMenu {
+	export class Model {
+		private height: number | string = 24
```
Comment (archanaatr) [inline material-select/src/dotMenu.ets]: Update all the dimentions to vp

File: material-select/src/menuOptions.ets [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+		this.checked = Boolean(checked)
+		return this
+	}
+
+	getChecked(): boolean {
+		return this.checked
+	}
+}
+
+class DotMenuOption {
```
Comment (archanaatr) [inline material-select/src/menuOptions.ets]: Check to extend from MenuOption

File: material-select/src/menuOptions.ets [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+	}
+
+	setIcon(icon: string | Resource) {
+		if (icon == null || icon == undefined) {
+			icon = ''
+		}
+		this.icon = icon
+	}
+
+	getIcon(): string | Resource {
```
Comment (archanaatr) [inline material-select/src/menuOptions.ets]: Recheck code formatting, current formatting have lot of space tabs.
Open each ets file > Ctrl + Alt + L

File: package.json [hunk 9]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "material_select",
```
Comment (archanaatr) [inline package.json]: Update the lib name as mentioned in the checklist

**Review by archanaatr — COMMENTED** [review]
please check the comments

File: package.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "@ohos/material-select",
```
Comment (archanaatr) [inline package.json]: keep a different name here

File: package.json [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+    "buildTool": "hvigor",
+    "directoryLevel": "project"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {
+    "@ohos/hvigor": "1.0.6",
+    "@ohos/hvigor-ohos-plugin": "1.0.6",
+    "@ohos/material-select": "file:material-select",
```
Comment (archanaatr) [inline package.json]: remove material-select dependency from main file

File: entry/src/main/ets/MainAbility/pages/index.ets [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+		new DotMenuOption(null, null, null),
+		new DotMenuOption(null, 1),
+		new DotMenuOption('Download', null),
+		new DotMenuOption('Download', null, null),
+		new DotMenuOption(null, null, $r('app.media.download')),
+		new DotMenuOption(null, 2, null),
+	]
+
+	aboutToAppear() {
+		this.selectModel.setTitle(null)
```
Comment (archanaatr) [inline entry/src/main/ets/MainAbility/pages/index.ets]: should avoid setting default values or if required provide a reset method in the model and call it

File: material-select/src/dotMenu.ets [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+				}, item => item.id)
+			}
+		}
+		.width(this.menuModel.getWidth())
+		.padding({ top: '7vp', bottom: '7vp' })
+		.borderRadius(this.menuModel.getBorderRadius())
+	}
+
+	build() {
+		Image(this.model.getIcon())
```
Comment (archanaatr) [inline material-select/src/dotMenu.ets]: Icon should be addedwithin the library? User should not be able to change it, please recheck and add

File: entry/src/main/ets/MainAbility/pages/index.ets [hunk 14]
```diff
@@ -0,0 +1,311 @@
...
+		this.multipleSelectMenu.push(new MultipleMenuOption(null, null, false))
+		this.multipleSelectMenu.push(new MultipleMenuOption(null, null, true))
+	}
+
+	build() {
+		Column() {
+			Column({ space: '10vp' }) {
+				Row({ space: '10vp' }) {
+					MaterialSelect({
+						menu: this.selectMenu,
```
Comment (archanaatr) [inline entry/src/main/ets/MainAbility/pages/index.ets]: recheck the code formatting, lot of tabs added to current code.

File: entry/src/main/ets/MainAbility/pages/index.ets [hunk 15]
```diff
@@ -0,0 +1,319 @@
...
+		this.multipleMenuModel.reset()
+		this.selectModel.reset()
+		this.selectMenuModel.reset()
+		this.dotModel.reset()
+		this.dotMenuModel.reset()
+	}
+
+	build() {
+		Column() {
+			Image('../../../../../../material-select/resources/dot_menu.png').height('100vp').width('100vp').alt($r("app.media.icon"))
```
Comment (archanaatr) [inline entry/src/main/ets/MainAbility/pages/index.ets]: It cannot be referred like this

File: material-select/src/dotMenu.ets [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+				}, item => item.id)
+			}
+		}
+		.width(this.menuModel.getWidth())
+		.padding({ top: '7vp', bottom: '7vp' })
+		.borderRadius(this.menuModel.getBorderRadius())
+	}
+
+	build() {
+		Image('../resources/dot_menu.png')
```
Comment (archanaatr) [inline material-select/src/dotMenu.ets]: Not like this, include in proper resource structure with in resources folder similar to folder structure in entry and refer using $r()

**Pull request merged**